### PR TITLE
Re-create containers in case hostname change

### PR DESCRIPTION
**Pull request by pyadvichuk**
From my investigation currently there is no way to set `hostname` defined in `config_db.json` to containers in automatic mode. 
`hostname` inside container can be updated in case restarting the container (by `docker rm -f`) however we're runing this only in case `hwsku` change. So, `sudo config reload -y` wil not affect hostname inside containers.
Need to pay an attention that some containers (like dhcp_relay) strongly required to have correct `hostname` as its use it for services runing inside.

**- What I did**
Set `hostname` inside each container from `<docker-name>.service`

**- How I did it**
Compare actual `hostname` from `config_db` and existing obtained by `docker inspect`

**- How to verify it**
- manually change 'hostname' in `config_db.json`
- run `sudo config reload -y`
- check hostname inside any container

**- Description for the changelog**
 Re-create containers in case hostname change

@lguohan @qiluo-msft: please review this

### CODE REVIEW

**Comment by akokhan** [comment]
@jleveque please review

File: files/build_templates/docker_image_ctl.j2 [hunk 1]
```diff
@@ -121,7 +128,7 @@ start() {
         fi
 
         # docker created with a different HWSKU, remove and recreate
```
Comment (jleveque) [inline files/build_templates/docker_image_ctl.j2]: Suggest update comment also: `# docker created with a different HWSKU and/or hostname, remove and recreate`

Comment (pyadvichuk) [reply files/build_templates/docker_image_ctl.j2]: done

**Comment by lguohan** [comment]
when we first load sonic, the default hostname is sonic, then when we load minigraph, the hostname changed, but I do not remember we need to recreate those containers. Now, if the hostname is changed, we have to recreate those container, it seems unnecessary.

**Comment by pyadvichuk** [comment]
> re-create containers in case of hostname change does not seem to be a good idea. it is better to implement something in hostcfgd to listen to hostname change and push the hostname into container.

so, should I decline this PR and just open the bug having issue description and steps to reproduce?

**Comment by jleveque** [comment]
>when we first load sonic, the default hostname is sonic, then when we load minigraph, the hostname changed, but I do not remember we need to recreate those containers. Now, if the hostname is changed, we have to recreate those container, it seems unnecessary.

I think we need to know what is different between these scenarios. Why do we not need to recreate containers after initially loading the minigraph, yet others are seeing this issue when changing the hostname?

File: files/build_templates/docker_image_ctl.j2 [hunk 2]
```diff
@@ -100,6 +105,7 @@ start() {
     # Obtain our HWSKU as we will mount directories with these names in each docker
     HWSKU=`sonic-cfggen -d -v 'DEVICE_METADATA["localhost"]["hwsku"]'`
     {%- endif %}
+    HOSTNAME=`sonic-cfggen -d -v 'DEVICE_METADATA["localhost"]["hostname"]'`
```
Comment (qiluo-msft) [inline files/build_templates/docker_image_ctl.j2]: >-d [](start = 27, length = 2)

-d will check Redis DB, which is inside a container. Can we just check -m Minigraph for less dependency?

Comment (pyadvichuk) [reply files/build_templates/docker_image_ctl.j2]: done

Comment (jleveque) [reply files/build_templates/docker_image_ctl.j2]: > Can we just check -m Minigraph for less dependency?

One concern I have with this approach is what if the minigraph on the device differs from the current config in the DB?

**Pull request closed without merging**